contracts/ocr2/src/contract.rs

```diff
@@ -344,18 +438,92 @@ pub fn execute_set_config(
             .add_attribute("config_count", config.config_count.to_string())
             .add_attributes(signers)
             .add_attributes(transmitters)
-            .add_attribute("f", f.to_string())
-            .add_attribute("onchain_config", Binary(onchain_calc).to_base64())
+            .add_attribute("f", proposal.f.to_string())
+            .add_attribute("onchain_config", Binary(onchain_config).to_base64())
             .add_attribute(
                 "offchain_config_version",
-                offchain_config_version.to_string(),
+                proposal.offchain_config_version.to_string(),
             )
-            .add_attribute("offchain_config", Binary(offchain_config).to_base64()),
+            .add_attribute("offchain_config", proposal.offchain_config.to_base64()),
     );
 
+    // Persist updated config
+    CONFIG.save(deps.storage, &config)?;
+
+    // Clear out proposal
+    PROPOSALS.remove(deps.storage, id.u128().into());
+
     Ok(response)
 }
 
+pub fn execute_set_config(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    id: ProposalId,
+    signers: Vec<Binary>,
+    transmitters: Vec<Addr>,
+    f: u8,
+    onchain_config: Vec<u8>,
+) -> Result<Response, ContractError> {
+    let is_owner = OWNER.is_owner(deps.as_ref(), &info.sender)?;
+    let mut proposal = PROPOSALS.load(deps.storage, id.u128().into())?;
+
+    // Only contract owner or proposal owner allowed
+    require!(is_owner || proposal.owner == info.sender, Unauthorized);
+
+    let signers_len = signers.len();
+
+    // validate new config
+    require!(f != 0, InvalidInput);
+    require!(signers_len <= MAX_ORACLES, TooManySigners);
+    require!(transmitters.len() == signers.len(), InvalidInput);
+    require!(3 * (usize::from(f)) < signers_len, InvalidInput);
+    require!(onchain_config.is_empty(), InvalidInput);
```

Comment:
This should probably be removed from here, looks like a relict from prev codebase.

It's unused in this fn, except for the check. And the check is incorrect as `onchain_config` is not empty anymore, but generated from contract init data (min, max), right?

Reply:
No, this was in fact re-added to match the behavior on Solidity: we expect the onchain config to be empty right now but it might be used in a future version. Then the code generates the expected v1 config internally and mixes it into the digest. https://github.com/smartcontractkit/libocr/blob/5b957c8342763a061fcbd6806ef98351f3cc7f93/contract2/OCR2Aggregator.sol#L220-L229

I agree that it's confusing and I'd rather remove the `onchain_config` parameter and directly use the `minAnswer`/`maxAnswer` in the digest calculation rather than calculate an intermediary value, wdyt? @kaleofduty